iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/ActivePowerControlImpl.java

```diff
@@ -43,22 +49,70 @@ public ActivePowerControlImpl(T component,
                                   boolean participate,
                                   double droop,
                                   double participationFactor,
-                                  double minPOverride,
-                                  double maxPOverride) {
+                                  double minTargetP,
+                                  double maxTargetP) {
         super(component);
         int variantArraySize = getVariantManagerHolder().getVariantManager().getVariantArraySize();
         this.participate = new TBooleanArrayList(variantArraySize);
         this.droop = new TDoubleArrayList(variantArraySize);
         this.participationFactor = new TDoubleArrayList(variantArraySize);
-        this.minPOverride = new TDoubleArrayList(variantArraySize);
-        this.maxPOverride = new TDoubleArrayList(variantArraySize);
-        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor, this.minPOverride, this.maxPOverride);
+        this.minTargetP = new TDoubleArrayList(variantArraySize);
+        this.maxTargetP = new TDoubleArrayList(variantArraySize);
+        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor, this.minTargetP, this.maxTargetP);
         for (int i = 0; i < variantArraySize; i++) {
             this.participate.add(participate);
             this.droop.add(droop);
             this.participationFactor.add(participationFactor);
-            this.minPOverride.add(minPOverride);
-            this.maxPOverride.add(maxPOverride);
+            this.minTargetP.add(checkTargetPLimit(minTargetP, "minTargetP", component));
+            this.maxTargetP.add(checkTargetPLimit(maxTargetP, "maxTargetP", component));
+        }
+        checkTargetPLimits(minTargetP, maxTargetP);
+    }
+
+    record PLimits(double minP, double maxP) { }
+
+    private PLimits getPLimits(T injection) {
+        double maxP = Double.MAX_VALUE;
+        double minP = -Double.MAX_VALUE;
+        if (injection instanceof Generator generator) {
+            maxP = generator.getMaxP();
+            minP = generator.getMinP();
+        }
+        if (injection instanceof Battery battery) {
+            maxP = battery.getMaxP();
+            minP = battery.getMinP();
+        }
+        return new PLimits(minP, maxP);
+    }
+
+    private double withinPMinMax(double value, T injection) {
+        PLimits pLimits = getPLimits(injection);
+
+        if (!Double.isNaN(value) && (value < pLimits.minP || value > pLimits.maxP)) {
```

Comment:
There is no need to add the `!Double.isNaN(value)` part in the condition:

> [For numerical comparison operators <, <=, >, and >=] If either operand is NaN, then the result is false.

([source](https://docs.oracle.com/javase/specs/jls/se17/html/jls-15.html#jls-15.20.1))

Idem for the other methods of this class.

Reply:
Done reluctantly.

Reply:
Seen with @vidaldid-rte: these checks are here to explicitly show that nothing should be done when a NaN value is encountered.